### ISSUE TITLE
[Feature Request] Custom Callback

### ISSUE THREAD

**Issue by cdiaz** [nestjs/passport#1]
By default, if authentication fails, the [AuthGuard](https://github.com/nestjs/passport/blob/master/lib/auth.guard.ts#L29) will respond with a generic  unauthorized error message.

But In some cases the built-in options are not sufficient to handle an authentication request. 
Then it should be possible provide a [Custom Callback](http://www.passportjs.org/docs/authenticate/) to allow the application to handle success or failure.
For example:

```js
const CustomCallback = (err, user, info) => {
  let message
  if (err) {
    return Promise.reject(err || new UnauthorizedException());
  } else if (typeof info != 'undefined' || !user) {
    switch (info.message) {
      case 'No auth token':
      case 'invalid signature':
      case 'jwt malformed':
      case 'invalid token':
        message = "You must provide a valid authenticated access token"
        break
      case 'jwt expired':
        message = "Your session has expired. Please log in again"
        break
    }
    return Promise.reject(new UnauthorizedException(message))
  }
  Promise.resolve(user);
}
```
```js
  @Get('data')
  @UseGuards(AuthGuard('jwt', { session: false }, CustomCallback))
  getSensitiveData() {
    // this route is restricted
  }
```

**Comment by kamilmysliwiec** [comment]
Fixed in `v1.0.5`
Example:
```typescript
@Get('data')
@UseGuards(AuthGuard('jwt', { session: false, callback }))
getSensitiveData() {
   ./ this route is restricted
}
```
when the `callback` looks like below:
```typescript
const callback = (err, user, info) => {
    if (err || !user) {
      throw err || new UnauthorizedException();
    }
    return user;
}
```

**Comment by cdiaz** [comment]
Thanks @kamilmysliwiec 

There is some way to define the callback globally to avoid having redundant code in each endpoint?
